.github/workflows/merge-release-tag.yml

```diff
@@ -0,0 +1,57 @@
+name: Merge Release Tag
+
+on:
+  push:
+    tags:
+      - 'mongosh@[0-9]+.[0-9]+.*'
+  workflow_dispatch:
+
+permissions:
+  contents: none # We use the github app to checkout and push changes
+
+jobs:
+  merge-release-tag:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: mongodb-js/devtools-shared/actions/setup-bot-token@main
+        id: app-token
+        with:
+          app-id: ${{ vars.DEVTOOLS_BOT_APP_ID }}
+          private-key: ${{ secrets.DEVTOOLS_BOT_PRIVATE_KEY }}
+
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.ref }}
+          fetch-depth: "0"
+          token: ${{ steps.app-token.outputs.token }}
+
+      - name: Extract version from tag
+        uses: actions-ecosystem/action-regex-match@d50fd2e7a37d0e617aea3d7ada663bd56862b9cc
+        id: version-match
+        with:
+          text: ${{ github.ref }}
+          regex: 'refs/tags/mongosh@([0-9]+\.[0-9]+\.*)'
+
+      - name: Validate release version
+        shell: bash
+        id: validate
+        run: |
+          if [[ -z "${{ steps.version-match.outputs.group1 }}" ]]; then
+            echo "Error: Could not extract version from tag"
+            exit 1
+          fi
+
+          if git rev-parse "release/v${{ steps.version-match.outputs.group1 }}" >/dev/null 2>&1; then
+            echo "Error: Branch release/v$RELEASE_TAG already exists"
+            echo "If this version has already been released consider using a different one."
+            exit 1
+          fi
```

Comment:
This seems redundant now, since we're not pushing that branch, there's no way for it to exist - is there?